server/data/data-library/src/test/java/org/apache/james/rrt/lib/AbstractRecipientRewriteTableTest.java

```diff
@@ -338,6 +338,34 @@ public void removeGroupMappingShouldDelete() throws Exception {
             .isEqualTo(MappingsImpl.empty());
     }
 
+    @Test
+    public void addAliasMappingShouldStore() throws Exception {
+        Domain domain = Domain.LOCALHOST;
+        String address2 = "test@james";
+        MappingSource source = MappingSource.fromUser(USER, domain);
+
+        virtualUserTable.addMapping(source, Mapping.alias(ADDRESS));
+        virtualUserTable.addMapping(source, Mapping.alias(address2));
+
+        assertThat(virtualUserTable.getMappings(USER, domain)).hasSize(2);
```

Comment:
To enforce storage without recursive resolution, we should tend to prefer **getUserDomainMappings**

See https://github.com/linagora/james-project/pull/2078

Could you do a commit before this PR for making add* & remove* test methods of this class using getUserDomainMappings, then in this commit also rely on this method?